src/pfx_ng_fmt_plug.c

```diff
@@ -149,7 +150,7 @@ static int valid(char *ciphertext, struct fmt_main *self)
 		goto bail;
 	if ((p = strtokm(NULL, "$")) == NULL) // stored_hmac
 		goto bail;
-	if (hexlenl(p) != 20*2)
+	if (hexlenl(p) != 20*2 && hexlenl(p) != 32*2)
 		goto bail;
```

Comment:
Here we simply use that 'wanted' len variable.  That way we validate that 20 bytes for sha1, 32 for sha256, 64 for sha512, etc.